File: problems/0027-remove-element/solution.cpp
class Solution {
public:
    void clearBack(vector<int>& nums, int val) {
        while(nums.size() > 0 && nums.back() == val) {
            nums.pop_back();
        }
    }
    
    int removeElement(vector<int>& nums, int val) {
        if (nums.size() == 0) {
            return 0;
        }
        if (nums.size() == 1 && nums[0] == val) {
            nums.pop_back();
            return 0;
        }
        
        
        clearBack(nums, val);
        
        int i = 0;
        while (i < nums.size() && nums.size() > 1) {
            if(nums[i] == val) {
                nums[i] = nums.back();
                nums.pop_back();
                clearBack(nums, val);   
            }
            i++;
        }
        return nums.size();
    }
};

File: problems/0027-remove-element/solution.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        
        j = 0
        for i in range(len(nums)):
            if nums[i] != val:
                # temp = nums[j]
                nums[j] = nums[i]
                # nums[i] = temp
                j += 1
        
        return j

